src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -601,13 +601,14 @@ private static ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e
                             syntax: node);
                     }
 
-                    constantValue = expression.ConstantValue;
+                    convertedExpression = BindToNaturalType(expression, diagnostics);
```

Comment:
Just curious, what scenario was affected?

Reply:
There were a couple of cases that were affected: matching against a constant interpolated string and matching against a switch expression I believe.